aria/modeling/service_instance.py

```diff
@@ -1685,12 +1725,31 @@ def inputs(cls):
     # endregion
 
     description = Column(Text)
+    relationship_edge = Column(Boolean)
     implementation = Column(Text)
+    configuration = Column(modeling_types.StrictDict(key_cls=basestring))
     dependencies = Column(modeling_types.StrictList(item_cls=basestring))
     executor = Column(Text)
     max_retries = Column(Integer)
     retry_interval = Column(Integer)
 
+    def configure(self):
+        from . import models
+        if (self.implementation is None) or (self.interface is None):
```

Comment:
oh, right. Please add a tiny comment next to it explaining that this case is only relevant for "Workflow operations" because I'm afraid I'll wonder about it the next time I see it as well :sweat_smile:  thanks!